File: src/main/java/io/vproxy/vfx/theme/ThemeFontProvider.java
package io.vproxy.vfx.theme;

import io.vproxy.vfx.manager.font.FontProvider;
import io.vproxy.vfx.manager.font.FontSettings;
import io.vproxy.vfx.manager.font.FontUsage;
import io.vproxy.vfx.manager.font.FontUsages;

public abstract class ThemeFontProvider implements FontProvider {
    @Override
    public void apply(FontUsage usage, FontSettings settings) {
        if (usage == FontUsages.windowTitle) {
            windowTitle(settings);
        } else if (usage == FontUsages.tableCellText) {
            tableCellText(settings);
        } else {
            _default(settings);
        }
    }

    protected abstract void windowTitle(FontSettings settings);

    protected abstract void tableCellText(FontSettings settings);

    protected abstract void _default(FontSettings settings);
}

File: src-ts/src/theme/ThemeFontProvider.ts
import type { FontProvider } from '../manager/font/FontProvider.js';
import type { FontSettings } from '../manager/font/FontSettings.js';
import type { FontUsage } from '../manager/font/FontUsage.js';
import { FontUsages } from '../manager/font/FontUsages.js';

export abstract class ThemeFontProvider implements FontProvider {
  apply(usage: FontUsage, settings: FontSettings): void {
    if (usage === FontUsages.windowTitle) {
      this.windowTitle(settings);
    } else if (usage === FontUsages.tableCellText) {
      this.tableCellText(settings);
    } else {
      this._default(usage, settings);
    }
  }

  protected abstract windowTitle(settings: FontSettings): void;
  protected abstract tableCellText(settings: FontSettings): void;
  protected abstract _default(usage: FontUsage, settings: FontSettings): void;
}
